x-pack/plugins/security_solution/public/resolver/view/panel.tsx

```diff
@@ -3,191 +3,283 @@
+       * | :--------------------- | :------------------------- | :----------------------  |
+       * | process detail         | entity_id of process       | null                     |
+       */
+      if (crumbEvent === '' && uiSelectedEvent) {
+        return 'processDetails';
+      }
+
+      /**
+       * | Crumb/Table            | &crumbId                   | &crumbEvent              |
+       * | :--------------------- | :------------------------- | :----------------------  |
+       * | relateds count by type | entity_id of process       | 'all'                    |
+       */
+
+      if (crumbEvent === 'all' && uiSelectedEvent) {
+        return 'eventCountsForProcess';
+      }
+
+      /**
+       * | Crumb/Table            | &crumbId                   | &crumbEvent              |
+       * | :--------------------- | :------------------------- | :----------------------  |
+       * | relateds list 1 type   | entity_id of process       | valid related event type |
+       */
+
+      if (crumbEvent in displayNameRecord && uiSelectedEvent) {
+        return 'processEventListNarrowedByType';
+      }
+    }
+
+    if (graphableProcessEntityIds.has(crumbEvent)) {
+      /**
+       * | Crumb/Table            | &crumbId                   | &crumbEvent              |
+       * | :--------------------- | :------------------------- | :----------------------  |
+       * | related event detail   | event_id of related event  | entity_id of process     |
+       */
+      return 'relatedEventDetail';
+    }
+
+    // The default 'Event List' / 'List of all processes' view
+    return 'processListWithCounts';
+  }, [uiSelectedEvent, crumbEvent, crumbId, graphableProcessEntityIds]);
+
+  useEffect(() => {
+    // dispatch `appDisplayedDifferentPanel` to sync state with which panel gets displayed
+    dispatch({
+      type: 'appDisplayedDifferentPanel',
+      payload: panelToShow,
+    });
+  }, [panelToShow, dispatch]);
+
+  const currentPanelView = useSelector(selectors.currentPanelView);
+
+  const panelInstance = useMemo(() => {
+    if (currentPanelView === 'processDetails') {
+      return (
+        <ProcessDetails processEvent={uiSelectedEvent!} pushToQueryParams={pushToQueryParams} />
+      );
+    }
+
+    if (currentPanelView === 'eventCountsForProcess') {
+      return (
+        <EventCountsForProcess
+          processEvent={uiSelectedEvent!}
+          pushToQueryParams={pushToQueryParams}
+          relatedStats={relatedStatsForIdFromParams!}
+        />
+      );
+    }
+
+    if (currentPanelView === 'processEventListNarrowedByType') {
+      return (
+        <ProcessEventListNarrowedByType
+          processEvent={uiSelectedEvent!}
+          pushToQueryParams={pushToQueryParams}
+          relatedStats={relatedStatsForIdFromParams!}
+          eventType={crumbEvent}
+        />
+      );
+    }
+
+    if (currentPanelView === 'relatedEventDetail') {
+      const parentCount: number = Object.values(
```

Comment:
Please comment this. Let's also schedule a ticket to move this to a model or selector. I'd like to avoid business logic here.